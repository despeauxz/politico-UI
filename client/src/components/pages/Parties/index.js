/* eslint-disable max-len */
import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import SideNav from '@containers/shared/SideNav';
import AuthHeader from '@components/shared/Header/AuthHeader';
import Modal from '@components/shared/Modal';
import '@components/shared/SideNav/index.scss';
import Plus from '@components/shared/Plus';
import Preloader from '@components/shared/Preloader/Preloader';
import NewPartyForm from '@containers/shared/Form/Party/NewParty';
import EditPartyForm from '@containers/shared/Form/Party/EditParty';
import DeletePartyForm from '@components/shared/Form/Party/DeleteParty';
import './index.scss';



class Parties extends Component {
    static propTypes = {
        user: PropTypes.object.isRequired,
        toggle: PropTypes.bool.isRequired,
        parties: PropTypes.array.isRequired,
        loading: PropTypes.bool,
        toggleModal: PropTypes.func.isRequired,
        getParties: PropTypes.func.isRequired
    }

    constructor(props) {
        super(props);

        this.state = {
            type: '',
            party: {}
        };
    }

    componentDidMount() {
        const { getParties } = this.props;
        getParties();
    }

    displayModal = () => {
        const { toggle } = this.props;
        const { type, party } = this.state;
        switch (type) {
        case 'create':
            return <Modal toggle={toggle} title="Add Party">
                <NewPartyForm />
            </Modal>;
        case 'edit':
            return <Modal toggle={toggle} title="Edit Party">
                <EditPartyForm party={party} />
            </Modal>;
        case 'delete':
            return <Modal toggle={toggle} title="Are you sure you want delete this Party?">
                <DeletePartyForm party={party} />
            </Modal>;
        default:
            return '';
        }
    }

    create = () => {
        const { toggleModal, toggle } = this.props;
        toggleModal(!toggle);
        this.setState({
            type: 'create'
        });
    }

    edit(party) {
        const { toggleModal, toggle } = this.props;
        toggleModal(!toggle);
        this.setState({
            type: 'edit',
            party
        });
    }

    delete(party) {
        const { toggleModal, toggle } = this.props;
        toggleModal(!toggle);
        this.setState({
            type: 'delete',
            party
        });
    }

    render() {
        const { user, parties, loading } = this.props;
        const modal = this.displayModal();

        return (
            <Fragment>
                {loading && <Preloader
                    type="page"
                    style="Triangle"
                    width={80}
                    height={80}
                    color="blue"
                />}
                <div className="font-sans antialiased h-screen flex">
                    <SideNav />
                    <div className="flex-1 flex flex-col bg-white overflow-hidden">
                        <AuthHeader />

                        <div className="px-6 py-4 flex-1 overflow-y-scroll">
                            <div className="center text-center">
                                {user.isAdmin &&
                                    <button className="p-3 rounded text-white bg-teal-600 hover:bg-teal-700 shadow-lg" onClick={this.create}>
                                        <span className="inline-block mr-2">
                                            <Plus />
                                        </span>New Party
                                    </button>
                                }
                            </div>
                            <div className="mt-8 flex flex-wrap justify-between">
                                {parties.map((party, index) => {
                                    return (
                                        <div key={index.toString()} className="my-1 px-1 w-full md:w-1/2 lg:my-4 lg:px-4 lg:w-1/2 mb-4 relative">
                                            {user.isAdmin &&
                                                <div className="absolute top-0 right-0 more">
                                                    <i className="icon ion-ios-more" />
                                                    <ul>
                                                        <li><a href="#" onClick={this.edit.bind(this, party)}>Edit</a></li>
                                                        <li><a href="#" onClick={this.delete.bind(this, party)}>Delete</a></li>
                                                    </ul>
                                                </div>
                                            }
                                            <div className="md:flex items-center bg-white shadow-lg rounded-lg p-6">
                                                <img className="h-16 w-16 md:h-20 md:w-20 rounded-full mx-auto md:mx-0 md:mr-6 mb-4 md:mb-0 lg:mb-0" src={party.logourl} />
                                                <div className="text-center md:text-left">
                                                    <h2 className="text-normal text-semibold">{party.name}</h2>
                                                    <div className="text-purple-500">{party.fullname}</div>
                                                    <div className="text-gray-600">{party.hqaddress}</div>
                                                </div>
                                            </div>
                                        </div>
                                    );
                                })}
                            </div>
                        </div>
                    </div>
                </div>
                { modal }
            </Fragment>
        );
    }
}

export default Parties;
