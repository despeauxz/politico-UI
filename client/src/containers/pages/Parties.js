import { connect } from 'react-redux';
import Parties from '@components/pages/Parties';
import { toggleModal } from '@actions/ui';
import { getParties } from '@actions/parties';

const mapStateToProps = state => ({
    isAuthenticated: state.auth.isAuthenticated,
    user: state.auth.user,
    toggle: state.ui.toggle,
    parties: state.party.parties,
    loading: state.party.loading
});

export default connect(mapStateToProps, { toggleModal, getParties })(Parties);