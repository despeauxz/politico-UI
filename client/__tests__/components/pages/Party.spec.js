import React from 'react';
import Parties from '@components/pages/Parties';

const props = {
    toggle: false,
    type: '',
    user: {
        isAdmin: true
    },
    parties: [],
    toggleModal: jest.fn(),
    getParties: jest.fn()
};

describe('Party Pages', () => {
    it('should render the index page based on props ~ type', () => {
        const shallowWrapper = shallow(<Parties {...props} />);

        expect(toJson(shallowWrapper)).toMatchSnapshot();
        expect(shallowWrapper.find('Parties')).toBeTruthy();
    });

    it('should render the index page and New party modal based on props ~ type', () => {
        const shallowWrapper = shallow(<Parties {...props} type="create" toggle={true} />);

        expect(toJson(shallowWrapper)).toMatchSnapshot();
        expect(shallowWrapper.find('NewPartyForm')).toBeTruthy();
    });

    it('should render the index page and Edit party modal based on props ~ type', () => {
        const shallowWrapper = shallow(<Parties {...props} type="edit" toggle={true} />);

        expect(toJson(shallowWrapper)).toMatchSnapshot();
        expect(shallowWrapper.find('EditPartyForm')).toBeTruthy();
    });

    it('should render the index page and Delete party modal based on props ~ type', () => {
        const shallowWrapper = shallow(<Parties {...props} type="delete" toggle={true} />);

        expect(toJson(shallowWrapper)).toMatchSnapshot();
        expect(shallowWrapper.find('DeletePartyForm')).toBeTruthy();
    });
});