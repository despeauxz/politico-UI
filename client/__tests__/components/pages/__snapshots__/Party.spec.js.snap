// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Party Pages should render the index page and Delete party modal based on props ~ type 1`] = `
<Fragment>
  <div
    className="font-sans antialiased h-screen flex"
  >
    <Connect(Sidebar) />
    <div
      className="flex-1 flex flex-col bg-white overflow-hidden"
    >
      <AuthHeader />
      <div
        className="px-6 py-4 flex-1 overflow-y-scroll"
      >
        <div
          className="center text-center"
        >
          <button
            className="p-3 rounded text-white bg-teal-600 hover:bg-teal-700 shadow-lg"
            onClick={[Function]}
          >
            <span
              className="inline-block mr-2"
            >
              <Plus />
            </span>
            New Party
          </button>
        </div>
        <div
          className="mt-8 flex flex-wrap justify-between"
        />
      </div>
    </div>
  </div>
</Fragment>
`;

exports[`Party Pages should render the index page and Edit party modal based on props ~ type 1`] = `
<Fragment>
  <div
    className="font-sans antialiased h-screen flex"
  >
    <Connect(Sidebar) />
    <div
      className="flex-1 flex flex-col bg-white overflow-hidden"
    >
      <AuthHeader />
      <div
        className="px-6 py-4 flex-1 overflow-y-scroll"
      >
        <div
          className="center text-center"
        >
          <button
            className="p-3 rounded text-white bg-teal-600 hover:bg-teal-700 shadow-lg"
            onClick={[Function]}
          >
            <span
              className="inline-block mr-2"
            >
              <Plus />
            </span>
            New Party
          </button>
        </div>
        <div
          className="mt-8 flex flex-wrap justify-between"
        />
      </div>
    </div>
  </div>
</Fragment>
`;

exports[`Party Pages should render the index page and New party modal based on props ~ type 1`] = `
<Fragment>
  <div
    className="font-sans antialiased h-screen flex"
  >
    <Connect(Sidebar) />
    <div
      className="flex-1 flex flex-col bg-white overflow-hidden"
    >
      <AuthHeader />
      <div
        className="px-6 py-4 flex-1 overflow-y-scroll"
      >
        <div
          className="center text-center"
        >
          <button
            className="p-3 rounded text-white bg-teal-600 hover:bg-teal-700 shadow-lg"
            onClick={[Function]}
          >
            <span
              className="inline-block mr-2"
            >
              <Plus />
            </span>
            New Party
          </button>
        </div>
        <div
          className="mt-8 flex flex-wrap justify-between"
        />
      </div>
    </div>
  </div>
</Fragment>
`;

exports[`Party Pages should render the index page based on props ~ type 1`] = `
<Fragment>
  <div
    className="font-sans antialiased h-screen flex"
  >
    <Connect(Sidebar) />
    <div
      className="flex-1 flex flex-col bg-white overflow-hidden"
    >
      <AuthHeader />
      <div
        className="px-6 py-4 flex-1 overflow-y-scroll"
      >
        <div
          className="center text-center"
        >
          <button
            className="p-3 rounded text-white bg-teal-600 hover:bg-teal-700 shadow-lg"
            onClick={[Function]}
          >
            <span
              className="inline-block mr-2"
            >
              <Plus />
            </span>
            New Party
          </button>
        </div>
        <div
          className="mt-8 flex flex-wrap justify-between"
        />
      </div>
    </div>
  </div>
</Fragment>
`;
